test/mpi/gentests.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/sh
+#
+# (C) 2018 by Argonne National Laboratory.
+#     See COPYRIGHT in top-level directory.
+#
+# Generate datatype testing build instructions for every line in
+# configuration file dtp-test-config.txt
+
+types=""     # basic MPI datatypes for type signature
+counts=""    # counts for type signature
+pathname=""  # test file pathname
+testdir=""   # name of test directory
+testname=""  # name of test
+builddir=""  # directory in which executable will be located
+dirs=""      # all directories containing DTPools tests
+args=""      # extra args
+timelimit="" # time limit for testlist.in
+procs=""     # number of processes to use in testlist.in
+srcdir=""    # relative directory in which this script is located
+maxbufsize=""
+other_args=""
+
+# help message
+show_help() {
+cat << EOF
+Usage: $0 [options]
+options:
+  --help    show this message and exit
+  --with-dtpools-datatypes=[typelist]
+            comma separated list of MPI Datatypes to use for
+            generating DTPools tests
+  --with-dtpools-maxbufsize=[bytes]
+            max buffer size of DTP objects
+EOF
+}
+
+while [ true ] ; do
+    if test -z "$1" ; then break ; fi
+    case $(echo $1 | cut -d= -f1) in
+        --help)
+            show_help
+            exit 1
+            ;;
+        --with-dtpools-datatypes)
+            types=$(echo $1 | cut -d= -f2)
+            shift
+            ;;
+        --with-dtpools-maxbufsize)
+            maxbufsize=$(echo $1 | cut -d= -f2)
+            shift
+            ;;
+    esac
+done
+
+# get source and build directories
+srcdir=`dirname $0`
+builddir=$PWD
+
+# reformat type list
+types=$(echo $types | tr "," " ")
+
+# set max count and large count
+MAXSIZE=1000
+MAXPINGSIZE=100
+MAXCOUNT=262144
+
+# NOTE: generate seeds for tests starting from '1'
+#       as srand() considers 'seed=0' and 'seed=1'
+#       equivalent: http://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/random_r.c;h=51a2e8c812aee78783bd6d38c1b6269d41c8e47e;hb=HEAD#l181
+seed=1
+
+while read -r line ; do
+    if [ ! `echo $line | head -c 1` = "#" ] ; then
+        # the line is not a comment
+	pathname=`echo $line | cut -f1 -d':'`
+        args=`echo $line | cut -f2 -d':'`
+        counts=`echo $line | cut -f3 -d':'`
+        timelimit=`echo $line | cut -f4 -d':'`
+        procs=`echo $line | cut -f5 -d':'`
+        mintestsize=`echo $line | cut -f6 -d':'`
+        maxtestsize=`echo $line | cut -f7 -d':'`
+        other_args=""
+    else
+        # the line is a comment
+        continue
+    fi
+
+    if [ -z "$pathname" ] ; then
+	echo "No pathname found"
```

Comment:
It looks like in my editor indentation is all messed up. Can you use spaces instead of tabs? Again, if you need I can fix this.

Reply:
I'll fix it.